src/dotnet/commands/dotnet-crossgen/Program.cs

```diff
@@ -0,0 +1,128 @@
+// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.IO;
+using Microsoft.DotNet.Cli.CommandLine;
+using Microsoft.DotNet.Cli.Utils;
+using Microsoft.DotNet.Tools.CrossGen.Exceptions;
+using Microsoft.DotNet.Tools.CrossGen.Outputs;
+namespace Microsoft.DotNet.Tools.CrossGen
+{
+    public static class CrossGenCommand
+    {
+        public static int Run(string[] args)
+        {
+            var app = new CommandLineApplication();
+            app.Name = "dotnet crossgen";
+            app.FullName = ".NET CrossGen Tool";
+            app.Description = "Tool for priming the dotnet optimization cache with crossgen'd assemblies";
+            app.HelpOption("-h|--help");
+
+            var appNameParam = app.Option("--appName <APPNAME>", "Name of the app", CommandOptionType.SingleValue);
+            var appRootParam = app.Option("--appRoot <DIR>", "App directory", CommandOptionType.SingleValue);
+
+            var outputDirParam = app.Option("--outputDir <DIR>", "(Optional) Directory to write the output cache to, default is DOTNET_HOSTING_OPTIMIZATION_CACHE", CommandOptionType.SingleValue);
+            var outputStructureParam = app.Option("--output-structure", "(Optional) Structure of the output. \"APP\" - Default option, obtain app's directory structure; \"CACHE\" - Arrange the crossgen'd assemblies in the structure of the optimization cache.", CommandOptionType.SingleValue);
+            var crossGenExeLocationParam = app.Option("--crossgenExe <FILE>", "(Optional) Location of crossgen executable.", CommandOptionType.SingleValue);
+            var generatePDBParam = app.Option("--generatePDB", "(Optional) Option to generate PDB", CommandOptionType.NoValue);
+            var diaSymReaderLocationParam = app.Option("--diasymreader <FILE>", "(Optional) Location of diasymreader", CommandOptionType.SingleValue);
+            var overwriteHashParam = app.Option("--overwrite-hash", "(Optional) If package hash value conflicts with existing cache, the program will exit unless this option is given.", CommandOptionType.NoValue);
```

Comment:
This is for the cache priming scenario. Say that if the hash value for a package on the deps file differs from the current hash value on the current optimization cache, the desired behavior should be to bail out of the crossgen process because there's something wrong with the assembly. If we overwrite this package cache, all other apps wouldn't be able to use this package's cache. (Come to think of it, maybe I should have left the hive entirely untouched. Currently the behavior is that every packages visited before the failed hash check would still be updated.)

What I had in mind is that if the customer want to choose to overwrite the entire package if there's a hash conflict. They can do so.